internal/docker/docker.go

```diff
@@ -190,3 +190,20 @@ func Copy(containerName, containerPath, localPath string) error {
 	}
 	return nil
 }
+
+// CheckDaemonRunning checks if the Docker daemon is running.
+// Returns nil if the daemon is running, or an error if it's not.
+func CheckDaemonRunning() error {
+	cmd := exec.Command("docker", "info")
+	errOutput := new(bytes.Buffer)
+	cmd.Stderr = errOutput
+
+	logger.Debugf("running command to check Docker daemon: %s", cmd)
```

Comment:
I'd use Trace level for this message

```suggestion
	logger.Tracef("running command to check Docker daemon: %s", cmd)
```